### PR TITLE
Startup Fixes

### LINKED ISSUE #1189: If DIE_AREA is set to wrong value, the flow errors out with strange error
### Description
DIE_AREA need to be set to value `0 0 34.5 57.12`, however when I accidently set it to wrong value "34.5 57.12" I got an error that does not make sense. (See logs below)

To reproduce: Take inverter design and replace DIE_AREA with incorrect value
```
set ::env(DIE_AREA) "34.5 57.12"
```

### Environment
```
python3 ./env.py issue-survey
Kernel: Linux v5.9.16-1-MANJARO
Distribution: manjaro 21.3.0
Python: v3.10.5 (OK)
Container Engine: docker v20.10.16 (OK)
OpenLane Git Version: merge-window-2-dev
pip: INSTALLED
pip:venv: INSTALLED
---
PDK Version Verification Status: OK
---
Git Log (Last 3 Commits)

2288f48 2022-07-08T17:28:00+04:00 Updated contributing to docs to include info about index.rst. Updated index.rst to include chip integraiton - Arman Avetisyan -  (HEAD -> merge-window-2, origin/merge-window-2)
fde55d3 2022-07-08T17:17:03+04:00 Removed wrongfully added files - Arman Avetisyan -  ()
c30ca75 2022-07-08T16:25:44+04:00 Updated according to feedback - Arman Avetisyan -  ()
```

Note: Log above  reproduced on my branch of OpenLane that has 0 modifications to the scripts, its only docs changed. Base commit used: f9b5781f5ef0bbdf39ab1c2bbd78be8db11b27f2

### Reproduction Material
[die_area_bug.tar.gz](https://github.com/The-OpenROAD-Project/OpenLane/files/9072629/die_area_bug.tar.gz)

### Expected behavior
A clear and concise error message for invalid config.

### Logs
```
[INFO]: Running Initial Floorplanning...
[ERROR]: during executing openroad script /openlane/scripts/openroad/floorplan.tcl
[ERROR]: Exit code: 1
[ERROR]: full log: designs/die_area_bug/runs/bug_tag/logs/floorplan/3-initial_fp.log
[ERROR]: Last 10 lines:
puts "\[INFO\]: Setting clock transition to: $::env(SYNTH_CLOCK_TRANSITION)"
[INFO]: Setting clock transition to: 0.15
set_clock_transition $::env(SYNTH_CLOCK_TRANSITION) [get_clocks $::env(CLOCK_PORT)]
[WARNING STA-0559] transition time can not be specified for virtual clocks.
puts "\[INFO\]: Setting timing derate to: [expr {$::env(SYNTH_TIMING_DERATE) * 10}] %"
[INFO]: Setting timing derate to: 0.5 %
set_timing_derate -early [expr {1-$::env(SYNTH_TIMING_DERATE)}]
set_timing_derate -late [expr {1+$::env(SYNTH_TIMING_DERATE)}]
Error: floorplan.tcl, 93 can't use empty string as operand of "-"
child process exited abnormally

```

### LINKED ISSUE #1323: flow.tcl not creating designs/design_name directory
## Description
As per document: https://github.com/The-OpenROAD-Project/OpenLane/blob/master/designs/README.md I'm trying to add new design with following command:
```
./flow.tcl -design <design_name> -init_design_config -add_to_designs
```
But its failed to add `design_name` directory
```
[ERROR]: Design spm_test not found.

```
Earlier its used to create directory automatically. But currently not.

If I'm missing anything please let me know.

If I create directory manually `OpenLane/designs/spm_test` and execute above command `config.json` added automatically.

## Expected behavior
Create `designs/<design_name` automatically.

## Environment

```
Kernel: Linux v3.10.0-1160.49.1.el7.x86_64
Distribution: centos 7
Python: v3.6.8 (OK)
Container Engine: docker v20.10.16 (OK)
OpenLane Git Version: d609968cca9ebc6aae6ba8ada9c16b1ed3ce4d0a
pip: INSTALLED
python-venv: INSTALLED
---
PDK Version Verification Status: OK
---
Git Log (Last 3 Commits)

d609968 2022-09-06T04:48:37+02:00 Replace mismatches' environment variables with flags (#1316) - Mohamed Gaber -  (HEAD -> master, origin/master, origin/HEAD)
2d97f88 2022-09-04T12:29:33+02:00 [BOT] Update magic (#1308) - Openlane Bot -  (tag: 2022.09.05)
460d3d8 2022-09-02T11:20:08+02:00 [BOT] Update yosys (#844) - Openlane Bot -  (tag: 2022.09.03)
---
Git Remotes

origin	https://github.com/The-OpenROAD-Project/OpenLane.git (fetch)
origin	https://github.com/The-OpenROAD-Project/OpenLane.git (push)

```

## Reproduction Material

Create new design with following command:

```
./flow.tcl -design spm_test -init_design_config -add_to_designs
```

## Logs

```
OpenLane Container (d609968):/openlane$ ./flow.tcl -design spm_test -init_design_config -add_to_designs
OpenLane d609968cca9ebc6aae6ba8ada9c16b1ed3ce4d0a
All rights reserved. (c) 2020-2022 Efabless Corporation and contributors.
Available under the Apache License, version 2.0. See the LICENSE file for more details.

[ERROR]: Design spm_test not found.

```

### PR DESCRIPTION
**Pull request by donn**
+ Added new script to import certain variables from `STD_CELL_LIBRARY_OPT` postfixed with `_OPT`
+ Added validation for `DIE_AREA` and `CORE_AREA`
~ Fixed `-init_design_config` by adjusting the location of the DESIGN_DIR check
---
Resolves #1189
Resolves #1323

### CODE REVIEW

File: scripts/tcl_commands/all.tcl [hunk 1]
```diff
@@ -223,10 +227,15 @@ proc trim_lib {args} {
     set_if_unset arg_values(-input) $::env(LIB_SYNTH_COMPLETE)
     set_if_unset arg_values(-output) $::env(LIB_SYNTH)
 
+    set no_synth_list "$::env(NO_SYNTH_CELL_LIST)"
+    if { [info exists $::env(NO_SYNTH_CELL_LIST)] } {
```
Comment (kareefardi) [inline scripts/tcl_commands/all.tcl:231]: is this `file exists` instead of `info exists`

Comment (kareefardi) [reply scripts/tcl_commands/all.tcl:231]: something else is not right here. i am not sure what

File: scripts/tcl_commands/all.tcl [hunk 2]
```diff
@@ -681,7 +709,14 @@ proc prep {args} {
...
                 set drc_exclude_list "$::env(DRC_EXCLUDE_CELL_LIST)"
             }
-            gen_exclude_list -lib resizer_opt -drc_exclude_list $drc_exclude_list -output $::env(synthesis_tmpfiles)/resizer_opt.exclude.list -drc_exclude_only -create_dont_use_list
+
+            puts_verbose $drc_exclude_list
+
+            gen_exclude_list \
+                -lib resizer_opt \
+                -drc_exclude_list $drc_exclude_list \
+                -drc_exclude_only \
```
Comment (kareefardi) [inline scripts/tcl_commands/all.tcl:718]: why is it `-drc_exlucde_only`? probably outside of this pr scope though

**Comment by kareefardi** [comment]
is there documentation for `_OPT` variables?

File: scripts/tcl_commands/all.tcl (hunk 1 above)
Comment (donn) [reply scripts/tcl_commands/all.tcl:231]: Yeah, the variable is wrong. oops.

File: scripts/tcl_commands/all.tcl (hunk 2 above)
Comment (donn) [reply scripts/tcl_commands/all.tcl:718]: I have no earthly idea

**Comment by donn** [comment]
The `_OPT` variables are the same as their non-OPT equivalent, but for `STD_CELL_LIBRARY_OPT`.

**Review by kareefardi — APPROVED** [review]

**Pull request merged**